src/server.rs

```diff
@@ -146,15 +145,41 @@ struct ServerStats {
     /// The count of errors writing to cache.
     pub cache_write_errors: u64,
     /// The number of successful cache writes.
-    pub cache_writes: u32,
+    pub cache_writes: u64,
     /// The total seconds spent writing cache entries.
-    pub cache_write_duration_s: u64,
-    /// The total nanoseconds spent writing cache entries.
-    pub cache_write_duration_ns: u32,
+    pub cache_write_duration: Duration,
+    /// The total time spent reading cache hits.
+    pub cache_read_hit_duration: Duration,
+    /// The total time spent reading cache misses.
+    pub cache_read_miss_duration: Duration,
     /// The count of compilation failures.
     pub compile_fails: u64,
 }
 
+impl Default for ServerStats {
+    fn default() -> ServerStats {
+        ServerStats {
+            compile_requests: u64::default(),
+            requests_unsupported_compiler: u64::default(),
+            requests_not_compile: u64::default(),
+            requests_not_cacheable: u64::default(),
+            requests_executed: u64::default(),
+            cache_errors: u64::default(),
+            cache_hits: u64::default(),
+            cache_misses: u64::default(),
+            non_cacheable_compilations: u64::default(),
+            forced_recaches: u64::default(),
+            cache_read_errors: u64::default(),
+            cache_write_errors: u64::default(),
+            cache_writes: u64::default(),
+            cache_write_duration: Duration::new(0, 0),
+            cache_read_hit_duration: Duration::new(0, 0),
+            cache_read_miss_duration: Duration::new(0, 0),
```

Comment:
I avoided doing this because I was too lazy to spell all this out, and `Duration` didn't have a `Default` implementation. [That'll be fixed in Rust 1.15](https://github.com/rust-lang/rust/issues/37546), but for now this is a reasonable thing to do.